src/lwc/geFormRenderer/geFormRenderer.js

```diff
@@ -1785,4 +1791,324 @@ export default class GeFormRenderer extends NavigationMixin(LightningElement) {
+            .find(recordTypeInfo =>
+                recordTypeInfo.name === opportunityRecordTypeName);
+        
+        if (!recordTypeInfo) return null;
+        return recordTypeInfo.recordTypeId;
+    }
+
+    get opportunityRecordTypeInfos() {
+        return this.opportunityObjectInfo &&
+            Object.values(this.opportunityObjectInfo.data.recordTypeInfos);
+    }
+
+    setDonationRecordTypeIdInFormState(opportunityRecordTypeId) {
+        const donationImportedRelatedRecordField =
+            relatedRecordFieldNameFor(DATA_IMPORT_DONATION_IMPORTED_FIELD.fieldApiName);
+        const relatedRecord =
+            this.getFieldValueFromFormState(donationImportedRelatedRecordField);
+
+        let updatedRecord;
+        if (relatedRecord) {
+            updatedRecord = Object.assign(
+                relatedRecord,
+                {recordTypeId: opportunityRecordTypeId});
+        } else {
+            updatedRecord = {recordTypeId: opportunityRecordTypeId};
+        }
+
+        this.updateFormState({
+            [donationImportedRelatedRecordField]: updatedRecord
+        })
+    }
+
+    siblingRecordTypeFieldFor(fieldMappingDevName) {
+        const fieldMappingToRecordTypeId =
+            Object.values(GeFormService.fieldMappings)
+                .filter(f =>
+                    f.Target_Object_Mapping_Dev_Name ===
+                    this.objectMappingDevNameFor(fieldMappingDevName))
+                .find(f => f.Target_Field_API_Name === 'RecordTypeId')
+
+        return fieldMappingToRecordTypeId &&
+            fieldMappingToRecordTypeId.Source_Field_API_Name;
+    }
+
+    objectMappingDevNameFor(fieldMappingDevName) {
+        const fieldMapping = GeFormService.getFieldMappingWrapper(fieldMappingDevName);
+        return fieldMapping && fieldMapping.Target_Object_Mapping_Dev_Name;
+    }
+
+    parentRecordFieldFor(fieldMappingDevName) {
+        const objMapping =
+            GeFormService.getObjectMapping(
+                this.objectMappingDevNameFor(fieldMappingDevName));
+
+        return objMapping && objMapping.Imported_Record_Field_Name;
+    }
+
+    donorType() {
+        return this.getFieldValueFromFormState(
+            DATA_IMPORT_DONATION_DONOR_FIELD.fieldApiName);
+    }
+
+    /**
+     * @description Pass in a DataImport field's api to get that field's current
+     * value from the formState object.
+     * @param fieldApiName
+     * @returns {*} Current value stored in formState for the passed-in field.
+     */
+    getFieldValueFromFormState(fieldApiName) {
+        return this.formState[fieldApiName];
+    }
+
+    flatten(obj) {
+        let flatObj = {};
+        for (const [key, value] of Object.entries(obj)) {
+            if (value !== null && value !== undefined && value.hasOwnProperty('value')) {
+                flatObj[key] = value.value;
+            } else {
+                flatObj[key] = value;
+            }
+        }
+        return flatObj;
+    }
```

Comment:
Would this be something to possibly move to utilCommon or another shared location?